src/lib/swapper/swappers/ArbitrumBridgeSwapper/getTradeQuote/getTradeQuote.ts

```diff
@@ -0,0 +1,225 @@
+import { Erc20Bridger, EthBridger, getL2Network } from '@arbitrum/sdk'
+import type { L1ToL2TransactionRequest } from '@arbitrum/sdk/dist/lib/dataEntities/transactionRequest'
+import { ethAssetId, ethChainId, fromAssetId } from '@shapeshiftoss/caip'
+import { isEvmChainId } from '@shapeshiftoss/chain-adapters'
+import type {
+  GetEvmTradeQuoteInput,
+  SingleHopTradeQuoteSteps,
+  TradeQuote,
+} from '@shapeshiftoss/swapper'
+import {
+  makeSwapErrorRight,
+  type SwapErrorRight,
+  SwapperName,
+  TradeQuoteError,
+} from '@shapeshiftoss/swapper'
+import { KnownChainIds } from '@shapeshiftoss/types'
+import type { Result } from '@sniptt/monads'
+import { Err, Ok } from '@sniptt/monads'
+import { getDefaultSlippageDecimalPercentageForSwapper } from 'constants/constants'
+import { BigNumber } from 'ethers5'
+import { v4 as uuid } from 'uuid'
+import { getAddress, isAddressEqual } from 'viem'
+import { arbitrum } from 'viem/chains'
+import { usdcAssetId } from 'components/Modals/FiatRamps/config'
+import { bn } from 'lib/bignumber/bignumber'
+import { getEthersV5Provider } from 'lib/ethersProviderSingleton'
+import { assertGetEvmChainAdapter, getFees } from 'lib/utils/evm'
+
+import { assertValidTrade } from '../utils/helpers'
+
+const usdcOnArbitrumAssetId = 'eip155:42161/erc20:0xaf88d065e77c8cc2239327c5edb3a432268e5831'
+
+const fetchTokenFallbackGasEstimates = () =>
+  // https://github.com/OffchainLabs/arbitrum-token-bridge/blob/d17c88ef3eef3f4ffc61a04d34d50406039f045d/packages/arb-token-bridge-ui/src/util/TokenDepositUtils.ts#L45-L51
+  // Use hardcoded gas estimate values
+  // Values set by looking at a couple of different ERC-20 deposits
+  bn(240_000)
+
+export async function getTradeQuote(
+  input: GetEvmTradeQuoteInput,
+): Promise<Result<TradeQuote, SwapErrorRight>> {
+  const {
+    chainId,
+    sellAsset,
+    buyAsset,
+    accountNumber,
+    affiliateBps,
+    potentialAffiliateBps,
+    supportsEIP1559,
+    receiveAddress,
+    sellAmountIncludingProtocolFeesCryptoBaseUnit,
+    sendAddress,
+  } = input
+  if (!(isEvmChainId(sellAsset.chainId) && isEvmChainId(buyAsset.chainId))) {
```

Comment:
use `assertValidTrade` here instead of evm check and then later supported chain check via assertion?

Reply:
c9b315ed97